kolibri/plugins/management/assets/src/vue/manage-class-page/class-delete-modal.vue

```diff
@@ -68,14 +62,10 @@
         required: true,
       },
     },
-    data() {
-      return {
-        error_message: '',
-      };
-    },
     methods: {
-      submit() {
-        // need delete logic here.
+      classDelete() {
+        this.deleteClass(this.classid);
+        this.$emit('close');
```

Comment:
I totally agree that it's unfortunate to have all these mutations in the same namespace, and that vuex 2 should help with that problem. However if we violate the flux pattern now and allow actions to directly control view state through promise callbacks, the vuex 2 upgrade will be more work.

> Also this pattern spreads simple logic to multiple locations

For better or worse, that's what flux does: it adds boilerplate, but it gets rid of callback, promise, and event-driven behaviors that make large applications hard to reason about.

As in the Zen of Python...

> Complex is better than complicated

Perhaps there are other alternatives that maintain flux but have fewer mutations?